dropwizard-client/src/main/java/io/dropwizard/client/DropwizardSSLConnectionSocketFactory.java

```diff
@@ -71,7 +71,8 @@ private SSLContext buildSslContext() throws SSLInitializationException {
         final SSLContext sslContext;
         try {
             final SSLContextBuilder sslContextBuilder = new SSLContextBuilder();
-            sslContextBuilder.useProtocol(configuration.getProtocol());
+            sslContextBuilder.setProtocol(configuration.getProtocol());
+            sslContextBuilder.setProvider(configuration.getProvider());
```

Comment:
Would it make sense to only set the provider if `configuration.getProvider()` wasn't `null`?

Reply:
There is no desperate need in it, because `Security.getProvider(null)` doesn't throw NPE, but you're right, better to check. Updated.